rules/CodeQuality/Rector/MethodCall/AssertCompareOnCountableWithMethodToAssertCountRector.php

```diff
@@ -0,0 +1,91 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Rector\PHPUnit\CodeQuality\Rector\MethodCall;
+
+use Countable;
+use PhpParser\Node;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Identifier;
+use PhpParser\Node\Name;
+use PHPStan\Type\ObjectType;
+use Rector\Rector\AbstractRector;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\PHPUnit\Tests\CodeQuality\Rector\MethodCall\AssertCompareOnCountableWithMethodToAssertCountRector\AssertCompareOnCountableWithMethodToAssertCountRectorTest
+ */
+class AssertCompareOnCountableWithMethodToAssertCountRector extends AbstractRector
+{
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+$this->assertSame(1, $countable->count());
+CODE_SAMPLE
+                ,
+                <<<'CODE_SAMPLE'
+$this->assertCount(1, $countable);
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<MethodCall|StaticCall>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [MethodCall::class, StaticCall::class];
+    }
+
+    /**
+     * @param MethodCall|StaticCall $node
+     */
+    public function refactor(Node $node): MethodCall|StaticCall|null
+    {
+        $class = $node instanceof StaticCall ? $node->class : $node->var;
+
+        if ($this->getType($class)->isSuperTypeOf(new ObjectType('PHPUnit\Framework\TestCase'))->no()) {
+            return null;
+        }
+
+        if (
+            ! $node->name instanceof Identifier ||
+            ($node->name->toLowerString() !== 'assertsame' && $node->name->toLowerString() !== 'assertequals')
+        ) {
+            return null;
+        }
+
+        $right = $node->getArgs()[1]
```

Comment:
`->isFirstClassCallable()` should be verified first and return null, then verify args total is 2

Reply:
I've added the argument count but I'm not sure what you mean by
> `->isFirstClassCallable()` should be verified first and return null

Could you clarify further?

Reply:
`->getArgs()` got warning when there is firstclass callable inside, see

https://github.com/nikic/PHP-Parser/blob/e3f223f623980cdc074f770eb3d54aafb3af5346/lib/PhpParser/Node/Expr/CallLike.php#L32

so needs verify early:

```php
if ( ! $node->isFirstCallable()) {
    return null;
}

Reply:
See example

https://github.com/rectorphp/rector-phpunit/blob/acfcef8cfbceefee966253ef5f1b2ec57bc10e22/rules/CodeQuality/Rector/MethodCall/AssertNotOperatorRector.php#L66

Reply:
@samsonasik thank you for explaining! Update made.